gammapy/maps/hpxnd.py

```diff
@@ -382,8 +384,9 @@ def stack(self, other, weights=None):
                 "Can only stack equivalent maps or cutout of the same map."
             )
 
-        data = other.quantity.to_value(self.unit)
-
+        data = other.quantity.to_value(self.unit).copy()
```

Comment:
Could avoid the copy here, if `nan_to_num = False`...

Reply:
Without the copy I have two tests failing with a message about "data is read only"

Reply:
I meant moving the `.copy()` into the `if` block...